### ISSUE TITLE
magento2 EE 2.1.6 get error below

### ISSUE THREAD

**Issue by michaeldyl520** [NikZh/magento2-multiple-layered-navigation#10]
```
1 exception(s):
Exception #0 (Exception): Notice: Undefined offset: -6 in /home/dyl/website/m216/vendor/magento/framework/Search/Dynamic/Algorithm.php on line 300

Exception #0 (Exception): Notice: Undefined offset: -6 in /home/dyl/website/m216/vendor/magento/framework/Search/Dynamic/Algorithm.php on line 300
#0 /home/dyl/website/m216/vendor/magento/framework/Search/Dynamic/Algorithm.php(300): Magento\Framework\App\ErrorHandler->handler(8, 'Undefined offse...', '/home/dyl/websi...', 300, Array)
#1 /home/dyl/website/m216/vendor/magento/framework/Search/Dynamic/Algorithm.php(177): Magento\Framework\Search\Dynamic\Algorithm->_findValueSeparator(1, Object(Magento\Framework\Search\Adapter\Mysql\Aggregation\Interval))
#2 /home/dyl/website/m216/vendor/magento/framework/Search/Dynamic/Algorithm/Improved.php(70): Magento\Framework\Search\Dynamic\Algorithm->calculateSeparators(Object(Magento\Framework\Search\Adapter\Mysql\Aggregation\Interval))
#3 /home/dyl/website/m216/vendor/magento/framework/Search/Adapter/Mysql/Aggregation/Builder/Dynamic.php(50): Magento\Framework\Search\Dynamic\Algorithm\Improved->getItems(Object(Magento\Framework\Search\Request\Aggregation\DynamicBucket), Array, Object(Magento\Framework\Search\Dynamic\EntityStorage))
#4 /home/dyl/website/m216/vendor/magento/framework/Search/Adapter/Mysql/Aggregation/Builder.php(94): Magento\Framework\Search\Adapter\Mysql\Aggregation\Builder\Dynamic->build(Object(Magento\CatalogSearch\Model\Adapter\Mysql\Aggregation\DataProvider\Interceptor), Array, Object(Magento\Framework\Search\Request\Aggregation\DynamicBucket), Object(Magento\Framework\DB\Ddl\Table))
#5 /home/dyl/website/m216/vendor/magento/framework/Search/Adapter/Mysql/Aggregation/Builder.php(73): Magento\Framework\Search\Adapter\Mysql\Aggregation\Builder->processAggregations(Object(Magento\Framework\Search\Request), Object(Magento\Framework\DB\Ddl\Table), Array)
#6 /home/dyl/website/m216/vendor/magento/framework/Search/Adapter/Mysql/Adapter.php(83): Magento\Framework\Search\Adapter\Mysql\Aggregation\Builder->build(Object(Magento\Framework\Search\Request), Object(Magento\Framework\DB\Ddl\Table), Array)
#7 /home/dyl/website/m216/vendor/magento/module-search/Model/SearchEngine.php(42): Magento\Framework\Search\Adapter\Mysql\Adapter->query(Object(Magento\Framework\Search\Request))
#8 /home/dyl/website/m216/vendor/magento/framework/Search/Search.php(72): Magento\Search\Model\SearchEngine->search(Object(Magento\Framework\Search\Request))
#9 /home/dyl/website/m216/vendor/magento/module-catalog-search/Model/ResourceModel/Fulltext/Collection.php(327): Magento\Framework\Search\Search->search(Object(Magento\Framework\Api\Search\SearchCriteria))
#10 /home/dyl/website/m216/vendor/magento/framework/Data/Collection/AbstractDb.php(338): Magento\CatalogSearch\Model\ResourceModel\Fulltext\Collection->_renderFiltersBefore()
#11 /home/dyl/website/m216/vendor/magento/module-catalog-search/Model/ResourceModel/Fulltext/Collection.php(361): Magento\Framework\Data\Collection\AbstractDb->_renderFilters()
#12 /home/dyl/website/m216/vendor/magento/module-catalog-search/Model/ResourceModel/Fulltext/Collection.php(399): Magento\CatalogSearch\Model\ResourceModel\Fulltext\Collection->_renderFilters()
#13 /home/dyl/website/m216/var/generation/Niks/LayeredNavigation/Model/ResourceModel/Fulltext/Collection/Interceptor.php(154): Magento\CatalogSearch\Model\ResourceModel\Fulltext\Collection->getFacetedData('category')
#14 /home/dyl/website/m216/app/code/Niks/LayeredNavigation/Model/Layer/Filter/Category.php(126): Niks\LayeredNavigation\Model\ResourceModel\Fulltext\Collection\Interceptor->getFacetedData('category')
#15 /home/dyl/website/m216/vendor/magento/module-catalog/Model/Layer/Filter/AbstractFilter.php(199): Niks\LayeredNavigation\Model\Layer\Filter\Category->_getItemsData()
#16 /home/dyl/website/m216/vendor/magento/module-catalog/Model/Layer/Filter/AbstractFilter.php(156): Magento\Catalog\Model\Layer\Filter\AbstractFilter->_initItems()
#17 /home/dyl/website/m216/vendor/magento/module-catalog/Model/Layer/Filter/AbstractFilter.php(145): Magento\Catalog\Model\Layer\Filter\AbstractFilter->getItems()
#18 /home/dyl/website/m216/vendor/magento/module-catalog/Model/Layer/Category/AvailabilityFlag.php(33): Magento\Catalog\Model\Layer\Filter\AbstractFilter->getItemsCount()
#19 /home/dyl/website/m216/vendor/magento/module-catalog/Model/Layer/Category/AvailabilityFlag.php(23): Magento\Catalog\Model\Layer\Category\AvailabilityFlag->canShowOptions(Array)
#20 /home/dyl/website/m216/vendor/magento/module-layered-navigation/Block/Navigation.php(107): Magento\Catalog\Model\Layer\Category\AvailabilityFlag->isEnabled(Object(Niks\LayeredNavigation\Model\Layer\Category\Interceptor), Array)
#21 /home/dyl/website/m216/vendor/magento/module-layered-navigation-staging/Block/Navigation.php(51): Magento\LayeredNavigation\Block\Navigation->canShowBlock()
#22 /home/dyl/website/m216/vendor/magento/theme-frontend-luma/Magento_LayeredNavigation/templates/layer/view.phtml(17): Magento\LayeredNavigationStaging\Block\Navigation->canShowBlock()
#23 /home/dyl/website/m216/vendor/magento/framework/View/TemplateEngine/Php.php(59): include('/home/dyl/websi...')
#24 /home/dyl/website/m216/vendor/magento/framework/View/Element/Template.php(255): Magento\Framework\View\TemplateEngine\Php->render(Object(Magento\LayeredNavigationStaging\Block\Navigation), '/home/dyl/websi...', Array)
#25 /home/dyl/website/m216/vendor/magento/framework/View/Element/Template.php(279): Magento\Framework\View\Element\Template->fetchView('/home/dyl/websi...')
#26 /home/dyl/website/m216/vendor/magento/framework/View/Element/AbstractBlock.php(659): Magento\Framework\View\Element\Template->_toHtml()
#27 /home/dyl/website/m216/vendor/magento/framework/View/Layout.php(542): Magento\Framework\View\Element\AbstractBlock->toHtml()
#28 /home/dyl/website/m216/vendor/magento/framework/View/Layout.php(518): Magento\Framework\View\Layout->_renderBlock('catalog.leftnav')
#29 /home/dyl/website/m216/var/generation/Magento/Framework/View/Layout/Interceptor.php(206): Magento\Framework\View\Layout->renderNonCachedElement('catalog.leftnav')
#30 /home/dyl/website/m216/vendor/magento/framework/View/Layout.php(472): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('catalog.leftnav')
#31 /home/dyl/website/m216/var/generation/Magento/Framework/View/Layout/Interceptor.php(193): Magento\Framework\View\Layout->renderElement('catalog.leftnav', true)
#32 /home/dyl/website/m216/vendor/magento/framework/View/Layout.php(569): Magento\Framework\View\Layout\Interceptor->renderElement('catalog.leftnav')
#33 /home/dyl/website/m216/vendor/magento/framework/View/Layout.php(520): Magento\Framework\View\Layout->_renderContainer('sidebar.main')
#34 /home/dyl/website/m216/var/generation/Magento/Framework/View/Layout/Interceptor.php(206): Magento\Framework\View\Layout->renderNonCachedElement('sidebar.main')
#35 /home/dyl/website/m216/vendor/magento/framework/View/Layout.php(472): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('sidebar.main')
#36 /home/dyl/website/m216/var/generation/Magento/Framework/View/Layout/Interceptor.php(193): Magento\Framework\View\Layout->renderElement('sidebar.main', true)
#37 /home/dyl/website/m216/vendor/magento/framework/View/Layout.php(569): Magento\Framework\View\Layout\Interceptor->renderElement('sidebar.main')
#38 /home/dyl/website/m216/vendor/magento/framework/View/Layout.php(520): Magento\Framework\View\Layout->_renderContainer('div.sidebar.mai...')
#39 /home/dyl/website/m216/var/generation/Magento/Framework/View/Layout/Interceptor.php(206): Magento\Framework\View\Layout->renderNonCachedElement('div.sidebar.mai...')
#40 /home/dyl/website/m216/vendor/magento/framework/View/Layout.php(472): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('div.sidebar.mai...')
#41 /home/dyl/website/m216/var/generation/Magento/Framework/View/Layout/Interceptor.php(193): Magento\Framework\View\Layout->renderElement('div.sidebar.mai...', true)
#42 /home/dyl/website/m216/vendor/magento/framework/View/Layout.php(569): Magento\Framework\View\Layout\Interceptor->renderElement('div.sidebar.mai...')
#43 /home/dyl/website/m216/vendor/magento/framework/View/Layout.php(520): Magento\Framework\View\Layout->_renderContainer('columns')
#44 /home/dyl/website/m216/var/generation/Magento/Framework/View/Layout/Interceptor.php(206): Magento\Framework\View\Layout->renderNonCachedElement('columns')
#45 /home/dyl/website/m216/vendor/magento/framework/View/Layout.php(472): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('columns')
#46 /home/dyl/website/m216/var/generation/Magento/Framework/View/Layout/Interceptor.php(193): Magento\Framework\View\Layout->renderElement('columns', true)
#47 /home/dyl/website/m216/vendor/magento/framework/View/Layout.php(569): Magento\Framework\View\Layout\Interceptor->renderElement('columns')
#48 /home/dyl/website/m216/vendor/magento/framework/View/Layout.php(520): Magento\Framework\View\Layout->_renderContainer('main.content')
#49 /home/dyl/website/m216/var/generation/Magento/Framework/View/Layout/Interceptor.php(206): Magento\Framework\View\Layout->renderNonCachedElement('main.content')
#50 /home/dyl/website/m216/vendor/magento/framework/View/Layout.php(472): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('main.content')
#51 /home/dyl/website/m216/var/generation/Magento/Framework/View/Layout/Interceptor.php(193): Magento\Framework\View\Layout->renderElement('main.content', true)
#52 /home/dyl/website/m216/vendor/magento/framework/View/Layout.php(569): Magento\Framework\View\Layout\Interceptor->renderElement('main.content')
#53 /home/dyl/website/m216/vendor/magento/framework/View/Layout.php(520): Magento\Framework\View\Layout->_renderContainer('page.wrapper')
#54 /home/dyl/website/m216/var/generation/Magento/Framework/View/Layout/Interceptor.php(206): Magento\Framework\View\Layout->renderNonCachedElement('page.wrapper')
#55 /home/dyl/website/m216/vendor/magento/framework/View/Layout.php(472): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('page.wrapper')
#56 /home/dyl/website/m216/var/generation/Magento/Framework/View/Layout/Interceptor.php(193): Magento\Framework\View\Layout->renderElement('page.wrapper', true)
#57 /home/dyl/website/m216/vendor/magento/framework/View/Layout.php(569): Magento\Framework\View\Layout\Interceptor->renderElement('page.wrapper')
#58 /home/dyl/website/m216/vendor/magento/framework/View/Layout.php(520): Magento\Framework\View\Layout->_renderContainer('root')
#59 /home/dyl/website/m216/var/generation/Magento/Framework/View/Layout/Interceptor.php(206): Magento\Framework\View\Layout->renderNonCachedElement('root')
#60 /home/dyl/website/m216/vendor/magento/framework/View/Layout.php(472): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('root')
#61 /home/dyl/website/m216/var/generation/Magento/Framework/View/Layout/Interceptor.php(193): Magento\Framework\View\Layout->renderElement('root', true)
#62 /home/dyl/website/m216/vendor/magento/framework/View/Layout.php(938): Magento\Framework\View\Layout\Interceptor->renderElement('root')
#63 /home/dyl/website/m216/vendor/magento/framework/Interception/Interceptor.php(146): Magento\Framework\View\Layout->getOutput()
#64 /home/dyl/website/m216/var/generation/Magento/Framework/View/Layout/Interceptor.php(494): Magento\Framework\View\Layout\Interceptor->___callPlugins('getOutput', Array, Array)
#65 /home/dyl/website/m216/vendor/magento/framework/View/Result/Page.php(243): Magento\Framework\View\Layout\Interceptor->getOutput()
#66 /home/dyl/website/m216/vendor/magento/framework/View/Result/Layout.php(164): Magento\Framework\View\Result\Page->render(Object(Magento\Framework\App\Response\Http\Interceptor))
#67 /home/dyl/website/m216/vendor/magento/framework/Interception/Interceptor.php(74): Magento\Framework\View\Result\Layout->renderResult(Object(Magento\Framework\App\Response\Http\Interceptor))
#68 /home/dyl/website/m216/vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Framework\View\Result\Page\Interceptor->___callParent('renderResult', Array)
#69 /home/dyl/website/m216/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'renderResult', Object(Magento\Framework\View\Result\Page\Interceptor), Array, 'result-varnish-...')
#70 /home/dyl/website/m216/vendor/magento/module-page-cache/Model/Controller/Result/VarnishPlugin.php(74): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Response\Http\Interceptor))
#71 /home/dyl/website/m216/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\PageCache\Model\Controller\Result\VarnishPlugin->aroundRenderResult(Object(Magento\Framework\View\Result\Page\Interceptor), Object(Closure), Object(Magento\Framework\App\Response\Http\Interceptor))
#72 /home/dyl/website/m216/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'renderResult', Object(Magento\Framework\View\Result\Page\Interceptor), Array, 'result-builtin-...')
#73 /home/dyl/website/m216/vendor/magento/module-page-cache/Model/Controller/Result/BuiltinPlugin.php(67): Magento\Framework\View\Result\Page\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Response\Http\Interceptor))
#74 /home/dyl/website/m216/vendor/magento/framework/Interception/Interceptor.php(142): Magento\PageCache\Model\Controller\Result\BuiltinPlugin->aroundRenderResult(Object(Magento\Framework\View\Result\Page\Interceptor), Object(Closure), Object(Magento\Framework\App\Response\Http\Interceptor))
#75 /home/dyl/website/m216/var/generation/Magento/Framework/View/Result/Page/Interceptor.php(130): Magento\Framework\View\Result\Page\Interceptor->___callPlugins('renderResult', Array, Array)
#76 /home/dyl/website/m216/vendor/magento/framework/App/Http.php(139): Magento\Framework\View\Result\Page\Interceptor->renderResult(Object(Magento\Framework\App\Response\Http\Interceptor))
#77 /home/dyl/website/m216/vendor/magento/framework/App/Bootstrap.php(258): Magento\Framework\App\Http->launch()
#78 /home/dyl/website/m216/index.php(39): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http))
#79 {main}
```
How do I fixed it?

**Comment by NikZh** [comment]
I'll try to fix this problem tonight (CET)

**Comment by NikZh** [comment]
This issue already fixed at 0.0.4 version
